File: toluaxx/src/tests/tarray.cxx
#include "tarray.hxx"

int a[10] = {0,1,2,3,4,5,6,7,8,9};
Point p[10] = {{0,1},{1,2},{2,3},{3,4},{4,5},{5,6},{6,7},{7,8},{8,9},{9,10}};
Point* pp[10];

int ma[10];
Point mp[10];
Point* mpp[10];

Array array;
Array* parray = &array;

void init(){
  for(int i=0;i<10;++i){
    pp[i] = &p[i];
    
    ma[i] = a[i];
    mp[i] = p[i];
    mpp[i] = pp[i];
    
    array.a[i] = a[i];
    array.p[i] = p[i];
    array.pp[i] = pp[i];
  }
}

File: toluaxx/src/tests/tarray.lua
require"base"
require"marray"

init()

local t=test("Array test")
--local t=tolua.test("","Array test","")

for i=0,9 do
   t(a[i],i,"a[i]==i i="..tostring(i))
end

for i=1,9 do
   t(p[i-1].y,p[i].x,"p[i-1].y==p[i].x i="..tostring(i))
end

for i=0,9 do
   t(M.a[i],i,"M.a[i]==i i="..tostring(i))
end

for i=1,9 do
   t(M.p[i-1].y,M.p[i].x,"M.p[i-1].y==M.p[i].x i="..tostring(i))
end

for i=0,9 do
   t(pp[i].x,M.p[i].x,"pp[i].x==M.p[i].x i="..tostring(i))
   t(p[i].y,M.pp[i].y,"p[i].y==M.pp[i].y i="..tostring(i))
end

for i=0,9 do
   t(array.a[i],parray.a[i],"array.a[i]==parray.a[i] i="..tostring(i))
   t(array.p[i].x,parray.pp[i].x,"array.p[i].x==parray.pp[i].x i="..tostring(i))
   t(array.p[i].y,parray.pp[i].y,"array.p[i].y==parray.pp[i].y i="..tostring(i))
end

for i=0,9 do
   array.a[i]=a[9-i]
   M.a[i]=9-i
   t(array.a[i],M.a[i],"array.a[i]==M.a[i] i="..tostring(i))
end

for i=1,9 do
   array.p[i] = array.pp[0]
   t(array.p[i].x,0,"array.p[i].x==0 i="..tostring(i))
   t(array.p[i].y,1,"array.p[i].y==1 i="..tostring(i))
end

--t:log()
test()
--print(t.result)
--print(t.errors)
--print(t.report)

